lib/saml_idp/request.rb

```diff
@@ -159,6 +160,13 @@ def document
     end
     private :document
 
+    def name_id_format_node
+      @_name_id_format_node ||= xpath('//samlp:AuthnRequest/samlp:NameIDPolicy/@Format',
```

Comment:
if this could be nil, `@var ||=` is not a good way to memoize IMO, we need to use `defined?` because a defined-but-nil value is still a case we want to avoid walking the XML doc again:

as in:

```ruby
return @_name_id_format_node if defined?(@_name_id_format_node)
@_name_id_format_node = xpath('//samlp:AuthnRequest/samlp:NameIDPolicy/@Format', ...)
```

Reply:
done. thanks